provider/aws/cloudformation.go

```diff
@@ -27,7 +33,10 @@ type cloudformationStackManager struct {
 	dryrunPath        string
 	skipVersionCheck  bool
 	cfnAPI            cloudformationiface.CloudFormationAPI
+	s3API             s3iface.S3API
 	ec2API            ec2iface.EC2API
+	ecsAPI            ecsiface.ECSAPI
```

Comment:
remove

Reply:
I can't remove ecsAPI or ecrAPI because they are created from a `sess` object that is only available from `newStackManager` and its only caller `InitializeContext` in init.go.

Alternate suggestion: I can replace ALL of the `*.*API` types with reference to `sess`, and have everything in this file create those connections as they are needed, instead of all-up-front, like it does now.  This move would also eliminate connections to these service when they are not needed. 

What do you think?